evap/rewards/tools.py

```diff
@@ -59,6 +60,21 @@ def reward_points_of_user(user):
 def is_semester_activated(semester):
     return SemesterActivation.objects.filter(semester=semester, is_active=True).exists()
 
+
+def target_points(progress, thresholds=None):
+    """
+    How many points should a user have based on their evaluation progress?
+    Progress should be 0.0 when none and 1.0 when all courses are evaluated.
+    thresholds is a list of tuples of form (<progress threshold>, <target reward value>).
+    If thresholds is None the REWARD_POINTS setting is used.
+    """
+    thresholds = thresholds if thresholds is not None else settings.REWARD_POINTS
+    # Filter reward point targets we have enough progress for
+    threshold_passed_reward_amount = map(lambda a: a[1], filter(lambda v: v[0] <= progress, thresholds))
```

Comment:
I think this
```
reached_thresholds = filter(lambda threshold: threshold[0] <= progress, thresholds)
threshold_passed_reward_amount = map(lamdba threshold: threshold[1], reached_thresholds)
```
drastically increases readability. Since I personally work with list comprehensions much more often than with map and filter, I'd even go for
```
passed_reward_amounts = [threshold[1] for threshold if threshold[0] <= progress]
```
This may even be short enough to directly wrap into the max:
```
reward_points = max([threshold[1] for threshold if threshold[0] <= progress], default=0)
```

@janno42, @karyon?

Reply:
I prefer the second option. Has the right amount of "things going on" per line at my opinion.

Reply:
Changed to unpacking in list comprehension in the newest commit.